### PR TITLE
To get the recommended unit file directory to place the pbs.service file for systemctl command

### PR DESCRIPTION
**Pull request by Bhagat-Rajput**
#### Bug/feature Description
* Unable to place the systemd unit dir file (**i.e. pbs.service**) into the recommended folder on ubuntu/debian machines.

#### Affected Platform(s)
* Ubuntu/Debian

#### Cause / Analysis / Design
* Placing the file (pbs.service ) into **/usr/lib/systemd/system/** directory on Ubuntu/Debian machines but this path is not documented anywhere.

#### Solution Description
* According to systemd manual page(**man systemd.unit**) PBS should place the unit dir file into **/lib/systemd/system/** folder on Ubuntu/debian machines.
* Added a macro to set the unit dir path based on machine type. 

#### Testing logs/output
* [centos_systemctl_log.txt](https://github.com/PBSPro/pbspro/files/2788749/centos_systemctl_log.txt)
* [ptl_pbs_systemd.txt](https://github.com/PBSPro/pbspro/files/2788750/ptl_pbs_systemd.txt)
* [ubuntu_systemctl_log.txt](https://github.com/PBSPro/pbspro/files/2788751/ubuntu_systemctl_log.txt)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [x] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: m4/with_unitdir.m4 [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+#
+
+AC_DEFUN([PBS_AC_WITH_UNITDIR],
+[
+  AC_ARG_WITH([unitdir],
+    AS_HELP_STRING([--with-unitdir=DIR],
```
[the remark this answers is not in the document]
Comment (Bhagat-Rajput) [reply m4/with_unitdir.m4]: Thanks @hirenvadalia for the review.
OS vendors defined the other paths as well i.e.
    /etc/systemd/system for Local configuration
    /run/systemd/system for Runtime units
    /usr/lib/systemd/system for Units of installed packages ..etc
So with the help of "--with-unitdir" user can place the file wherever user wants because i couldn't find any reason to restrict the unit file to one directory only.

File: m4/with_unitdir.m4 [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      [Specify the directory where you want to place the systemd unit file.]
+    )
+  )
+  AC_MSG_CHECKING([whether specific directory requested for systemd unit file])
+  AS_IF([test "x$with_unitdir" != "x"],
+  [systemd_dir="$with_unitdir"
+    AC_MSG_RESULT([yes])
+    AC_MSG_CHECKING([user specified path for systemd unit dir])],
+    [AC_MSG_RESULT([no])
+    systemd_dir="/usr/lib/systemd/system"
```
[the remark this answers is not in the document]
Comment (Bhagat-Rajput) [reply m4/with_unitdir.m4]: That we are checking in pbspro.spec.in file.

**Comment by subhasisb** [comment]
I feel (and agree with Hiren) that the user does not need such an option at configure time. The unit pat is a system dependency and if anything, we can enhance the code that finds out where to install it automatically.

File: m4/with_unitdir.m4 (hunk 1 above)
[the remark this answers is not in the document]
Comment (nithinj) [reply m4/with_unitdir.m4]: >  if in future OS vendors change those paths then anyways we have to change this m4 file to adopt new path, right?
You are right. IMHO, Until we update the code to adopt to the changes, a user who is not familiar with the semantics of m4 can make use of this harmless piece of code.

[the remark this answers is not in the document]
Comment (hirenvadalia) [reply m4/with_unitdir.m4]: @nithinj if the user is not familiar with m4 then they can simply copy the file to correct location. And my view this is a harmful code as it creates confusion.

**Comment by Bhagat-Rajput** [comment]
Thanks @subhasisb , @hirenvadalia  and @nithinj for sharing your views and based on your feedback have removed the "--with-unitdir" option.Please review it again.

File: pbspro.spec.in [hunk 3]
```diff
@@ -66,7 +66,7 @@
 %define pbs_dist %{pbs_name}-%{pbs_version}.tar.gz
 
 %if !%{defined _unitdir}
-%define _unitdir /usr/lib/systemd/system
+%define _unitdir @_unitdir@
```
Comment (mike0042) [inline pbspro.spec.in]: Please apply this change to pbspro.spec as well. We need to keep them in sync.

File: pbspro.spec [hunk 4]
```diff
@@ -513,7 +513,6 @@ fi
 %exclude %{pbs_prefix}/lib*/python/pbs_bootcheck*
 %exclude %{pbs_prefix}/libexec/install_db
 %exclude %{pbs_prefix}/libexec/pbs_habitat
-%exclude %{pbs_prefix}/libexec/pbs_schema_upgrade
```
Comment (Bhagat-Rajput) [inline pbspro.spec]: This change is not related to this PR. I think someone else has deleted the line but forgot to update the pbspro.spec file.

File: pbspro.spec.in (hunk 3 above)
Comment (Bhagat-Rajput) [reply pbspro.spec.in]: Thanks @mike0042 , changes added to the pbspro.spec file as well. Please review it.

File: pbspro.spec (hunk 4 above)
Comment (mike0042) [reply pbspro.spec]: The same change is present in #905. Depending on which PR gets merged first, the other will already have the change in place.
